### DISCUSSION TITLE
asCpScreen controller method - allow removing form from page completly

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#13593] [Ideas] [answered]
`asCpScreen` is a great method in craft control panel controllers. It allows us to define things like breadcrumbs or page title with chaining methods, instead of awkwardly declaring Twig variables in templates.

The problem is, that every page using this method has its contents enclosed in the form which can be submitted. Would be great if we could just get rid of this form and use `asCpScreen` for things like listing pages.

**Comment by piotrpog** [comment]
Just found out that this can be done by setting `->action()` method to false.

**Reply by brandonkelly** [reply]
It should be `null` by default, which would prevent it from being a form on full pages, however slideouts will still assume it should be a `<form>` so that’s something we should address in Craft 5.

**Comment by brandonkelly** [comment, marked as answer]
(Finally) made it possible to create non-`form` slideouts, for Craft 5.8 (2f925b33c9086e440706303bcf124f4862c9bc74).

To do it, you’ll need to include `containerElement: 'div'` in the `settings` object.

```js
const slideout = new Craft.CpScreenSlideout('my/render/action',
  containerElement: 'div',
});
```

When `containerElement` isn’t `'form'` (still the default), the slideout won’t get a “Save” button, and the close button will be labelled “Close” instead of “Cancel”.
